src/libraries/System.Private.CoreLib/src/System/IO/StreamReader.cs

```diff
@@ -187,18 +187,18 @@ public StreamReader(string path, bool detectEncodingFromByteOrderMarks)
         {
         }
 
-        public StreamReader(string path, Encoding encoding)
+        public StreamReader(string path, Encoding? encoding)
```

Comment:
I was suprised to see this change, as it was not approved as part of https://github.com/dotnet/runtime/issues/106236#issuecomment-2486568664.

But then I've found @jozkee comment https://github.com/dotnet/runtime/pull/106658#pullrequestreview-2246489688 referring to https://github.com/dotnet/runtime/issues/2376#issuecomment-580396637 that explains that we need to unify all ctors that accept `Encoding`. Which we already did for `StreamWriter` in https://github.com/dotnet/runtime/pull/106658. 

So everything is good 👍